### PR TITLE
Add new theme for social logins

### PR DESCRIPTION
**Pull request by daniel-ac-martin**
Adds a new theme specifically for when users should only log in with a
social provider.

This is just a first iteration and could be improved in the future with
some attention from a designer.

## Large
theme-for-social-providers-large

## Small
theme-for-social-providers-small

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
Robin has kindly updated it to be horizontal when the browser window is wide enough.
theme-for-social-providers-large2

File: sass/govuk_elements.scss [hunk 1]
```diff
@@ -18,3 +18,10 @@ $path: "../vendor/govuk_frontend_toolkit/images/";// Path to Frontend Toolkit im
 @import 'govuk-elements-sass/public/sass/elements/_panels';
 @import 'govuk-elements-sass/public/sass/elements/forms/_form-validation';
 @import 'govuk-elements-sass/public/sass/elements/forms/_form-block-labels';
+
+.list.horizontal li {
+  display: list-item;
```
Comment (paulwaitehomeoffice) [inline sass/govuk_elements.scss]: Good stuff — slightly surprised GOV.UK Elements didn’t have something like this already.

Do we need `display:list-item` here? It’s the user agent default for `<li>`s, right?

File: docker-compose.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ keycloak:
   volumes:
   - ./.docker-compose/keycloak.env:/etc/secrets/keycloak.env
   - ./govuk/:/opt/jboss/keycloak/themes/govuk/
+  - ./govuk-social-provider-login/:/opt/jboss/keycloak/themes/govuk-social-provider-login/
```
Comment (paulwaitehomeoffice) [inline docker-compose.yml]: Just to bike-shed the name, is either `govuk-social-provider-login-only` or `govuk-social-provider-only` marginally more descriptive?

**Comment by paulwaitehomeoffice** [comment]
I think this means we can remove `socialProviders` from messages_en.properties, right? I think that’s a message we added, I can’t see it in my Keycloak downloads.

File: docker-compose.yml (hunk 2 above)
Comment (daniel-ac-martin) [inline docker-compose.yml]: Yes I think you are right. The second one also has the benefit of being slightly shorter.

Naming things is hard!

**Comment by daniel-ac-martin** [comment]
No we still need it for when the standard govuk theme needs to show the social providers.

**Comment by daniel-ac-martin** [comment]
Felt the need to stick an 's' in there so it wasn't any shorter in the end but it is still clearer.

File: sass/govuk_elements.scss (hunk 1 above)
Comment (daniel-ac-martin) [inline sass/govuk_elements.scss]: Just had a play and I think you're right.

@RobinKnipe: What do you think?

File: docker-compose.yml (hunk 2 above)
Comment (paulwaitehomeoffice) [inline docker-compose.yml]: We’re just lucky we don‘t have any caches to invalidate or any risk of off-by-two errors.

**Comment by paulwaitehomeoffice** [comment]
> we still need it for when the standard govuk theme needs to show the social providers

Ah yes I see. Is there a case to be made for keeping that heading with the list on the social providers only theme too? If we did that, we could avoid having a different version of the login template in the social-only theme.

**Comment by daniel-ac-martin** [comment]
Unfortunately the headings are different sizes so we need to override it in favour of the h1 from `template.ftl`.

We could probably refactor the govuk theme in the future to reduce the code duplication.

**Comment by daniel-ac-martin** [comment]
Most of `template.ftl` is duplicated for example so it would be nice to get rid of that as well and the stuff in `login.ftl`.

**Comment by paulwaitehomeoffice** [comment]
> Unfortunately the headings are different sizes so we need to override it in favour of the `h1` from template.ftl.

Yeah, I was thinking the `<title>` and `<h1>` could, in theory, be “Sign in” like in govuk, and then the list keeps its own `<h2>`. But you’re the team who’s actually using the social sign-in, so I’m very happy to go with your design.

> We could probably refactor the govuk theme in the future to reduce the code duplication. Most of `template.ftl` is duplicated for example so it would be nice to get rid of that as well and the stuff in `login.ftl`.

Yeah I was hoping Keycloak would have some facility to disable everything except Identity Provider sign in, and that the templates could tell when that was in effect — then we wouldn’t need a separate theme for the social-login-only scenario. But as far as I can tell, it doesn’t currently have that.

**Comment by RobinKnipe** [comment]
After a chat with @pauldwaite, I'm going to merge the current changes, and test some of the alternatives discussed above with some of our users at the end of month...

**Pull request merged**